py/genv.py

```diff
@@ -23,7 +40,8 @@ def poll_pid(pid: int) -> bool:
     else:
         return True
 
-def poll_kernel(kernel_id: str) -> bool:
+
+def jupyter_kernel_kill(kernel_id: str) -> bool:
```

Comment:
this also does not kill the Jupyter kernel but checks if it is running
I like that you added the word `jupyter` here so wdyt about `poll_jupyter_kernel`?

Reply:
I agree, this is a better name.